plugin/pkg/scheduler/scheduler_test.go

```diff
@@ -170,164 +169,171 @@ func TestScheduler(t *testing.T) {
 	}
 }
 
-func TestSchedulerForgetAssumedPodAfterDelete(t *testing.T) {
-	// Set up a channel through which we'll funnel log messages from the watcher.
-	// This way, we can guarantee that when the test ends no thread will still be
-	// trying to write to t.Logf (which it would if we handed t.Logf directly to
-	// StartLogging).
-	ch := make(chan string)
-	done := make(chan struct{})
-	var wg sync.WaitGroup
-	wg.Add(1)
+func TestSchedulerNoPhantomPodAfterExpire(t *testing.T) {
+	stop := make(chan struct{})
+	defer close(stop)
+	queuedPodStore := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	scache := schedulercache.New(1*time.Second, stop)
+	scheduler, _, bindingChan, _ := setupTestSchedulerWithOnePod(t, queuedPodStore, scache)
+
+	waitExpireChan := make(chan struct{})
+	timeout := make(chan struct{})
+	go func() {
+		select {
+		case <-time.After(wait.ForeverTestTimeout):
+			close(timeout)
+		case <-waitExpireChan:
+		}
+	}()
 	go func() {
-		defer wg.Done()
 		for {
 			select {
-			case msg := <-ch:
-				t.Log(msg)
-			case <-done:
+			case <-timeout:
+				return
+			default:
+			}
+			pods, err := scache.List(labels.Everything())
+			if err != nil {
+				t.Fatalf("cache.List failed: %v", err)
+			}
+			if len(pods) == 0 {
+				close(waitExpireChan)
 				return
 			}
+			time.Sleep(1 * time.Second)
 		}
 	}()
-	eventBroadcaster := record.NewBroadcaster()
-	watcher := eventBroadcaster.StartLogging(func(format string, args ...interface{}) {
-		ch <- fmt.Sprintf(format, args...)
-	})
-	defer func() {
-		watcher.Stop()
-		close(done)
-		wg.Wait()
-	}()
-
-	// Setup stores to test pod's workflow:
-	// - queuedPodStore: pods queued before processing
-	// - scheduledPodStore: pods that has a scheduling decision
-	scheduledPodStore := cache.NewStore(cache.MetaNamespaceKeyFunc)
-	queuedPodStore := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	// waiting for the assumed pod to expire
+	select {
+	case <-waitExpireChan:
+	case <-timeout:
```

Comment:
can you get rid of the timeout goroutine and manage the timeout chan here with time.After?